src/default-config-backend.cpp

```diff
@@ -9,29 +9,61 @@
 #include <sys/inotify.h>
 #include <unistd.h>
 
-#define INOT_BUF_SIZE (1024 * sizeof(inotify_event))
-static char buf[INOT_BUF_SIZE];
-
+#define INOT_BUF_SIZE (sizeof(inotify_event) + NAME_MAX + 1)
 
 static std::string config_dir, config_file;
 wf::config::config_manager_t *cfg_manager;
 
-static void reload_config(int fd)
+static void readd_watch(int fd)
```

Comment:
why is this `readd_watch`? can we have a better name?

Reply:
I'm not sure, I think the name accurately describes what it does, and it's a function not used elsewhere, so I don't think it's a huge deal. I can change it if you have a better suggestion.

Reply:
oh, you're right, I just misread the name, I thought it was `read` + `d` + `_watch` but it is more like `re` + `add` + `watch`, which makes perfect sense indeed.